src/h3lib/lib/h3UniEdge.c

```diff
@@ -229,14 +241,26 @@ void H3_EXPORT(getH3UnidirectionalEdgeBoundary)(H3Index edge, GeoBoundary* gb) {
 
     int k = 0;
     for (int i = 0; i < origin.numVerts; i++) {
-        for (int j = 0; j < destination.numVerts; j++) {
-            if (geoAlmostEqualThreshold(&origin.verts[i], &destination.verts[j],
-                                        0.000001)) {
-                gb->verts[k].lat = origin.verts[i].lat;
-                gb->verts[k].lon = origin.verts[i].lon;
+        if (_hasMatchingVertex(&origin.verts[i], &destination)) {
+            // If we are on vertex 0, we need to handle the case where it's the
+            // end of the edge, not the beginning.
+            // TODO: This is fine for this limited case, but it might be
+            // better/cleaner to use a VertexGraph for this
+            printf("%d, %d\n", i,
+                   _hasMatchingVertex(&origin.verts[i + 1], &destination));
+            if (i == 0 &&
+                !_hasMatchingVertex(&origin.verts[i + 1], &destination)) {
+                postponedVertex = origin.verts[i];
+            } else {
+                gb->verts[k] = origin.verts[i];
                 k++;
             }
         }
     }
+    // If we postponed adding the last vertex, add it now
+    if (postponedVertex.lat && postponedVertex.lon) {
```

Comment:
This doesn't look safe, since (0,0) is a valid set of coordinates.

Reply:
True. Is there a better option here? The only thing I can think of is to add an additional var like `hasPostponedVertex`, which is pretty ugly.